lib/classes/package_detail_builder.rb

```diff
@@ -1,30 +1,34 @@
 class PackageDetailBuilder
   attr_reader :detail_type, :detail_params
 
-  def initialize(params)
+  def initialize(params, request_from_stockit)
     @detail_params = params["detail_attributes"]
     @detail_type = params["detail_type"]
+    @request_from_stockit = request_from_stockit
   end
 
+
   def build_detail
     return unless %w[computer electrical computer_accessory].include?(detail_type)
     klass = detail_type.classify.safe_constantize
-    params = lookup_ids_hash.merge(detail_params_without_id)
+    params = detail_params_without_some_params.merge(lookup_ids_hash)
     klass&.new(params)
   end
 
   private
 
-  def detail_params_without_id
+  def detail_params_without_some_params
```

Comment:
This is a strange name! I think it's fine to just call this method `detail_params` and make a comment inside the method that we are excluding various fields and why we are doing that